packages/mermaid/src/diagrams/common/common.ts

```diff
@@ -178,16 +178,18 @@ export const getMin = function (...values: number[]): number {
  * @returns The converted string
  */
 export const parseGenericTypes = function (text: string): string {
+  if (!text) {
+    return '';
+  }
+
   let cleanedText = text;
 
   if (text.split('~').length - 1 >= 2) {
     let newCleanedText = cleanedText;
 
-    // use a do...while loop instead of replaceAll to detect recursion
-    // e.g. Array~Array~T~~
     do {
       cleanedText = newCleanedText;
-      newCleanedText = cleanedText.replace(/~([^\s,:;]+)~/, '<$1>');
+      newCleanedText = cleanedText.replace(/~([^:;]+)~/, '<$1>');
```

Comment:
This regex will fail if you have more than 1 method parameter using generics. For example:

`Dictionary~Foo, Bar~, Wiz~Bang~` becomes `Dictionary<Foo, Bar<, Wiz>Bang>` when it should instead result in `Dictionary<Foo, Bar>, Wiz<Bang>`

That's the main reason I mentioned there's too much ambiguity using tildes as the syntax since there's not enough information to resolve whether this is 1 parameter or 2. (see #4555)

Reply:
I see that now. Added a test for that specific scenario.

The problem I see now is - we have been using ~~ to indicate a generic for a couple of years, and to change that would be breaking. Though I am willing to look at doing that, if we go into it with that understanding :)

I was thinking of having the code go through and keep track of where it is in the "parsing".  So that it would know if it had already processed a beginning or end bracket - but you need to be able to account for nested generics

Im open to other suggestions

Reply:
Yeah I thought about that, but the issue, like you mentioned, is that even if you scan from left to right, it's impossible to correctly pair up the matching ~~ without guessing. That's why using asymmetric tokens (e.g. <>, [], (), or {}) would be best.

The easiest way to facilitate that would be to replace any instance of <> with HTML-escaped angle brackets (`&lt;` and `&gt;`). This could be added as an alternative syntax without needing to remove support for the legacy tilde syntax and would unblock users wanting to take advantage of more complex generics.

Reply:
What I was working on today was going right to left.
Count how many tokens (~) there are, divide that by two (rounded down) to determine how many "sets" there are - assuming that if there is an odd number of tokens it would be the first one, and we will skip it.

Keeping track of which "set" we are in, and how many tokens are left, we can figure out if we want > or <.

This has not been nearly as clean as I thought - and I am going to scrap it, I think

Reply:
Resolving since this is a dupe of another comment above